tensorflow/core/ops/array_ops.cc

```diff
@@ -2292,6 +2294,23 @@ Status SliceHelper(InferenceContext* c, ShapeHandle begin_value,
   return Status::OK();
 }
 
+// This SliceHelper processes the output shape of the `slice`
+// when the tensor of `sizes` is not available (null).
+// In this situation, we could still obtain partial information
+// from `MakeShapeFromShapeTensor` as a best effort operation.
+Status SliceHelper(InferenceContext* c, ShapeHandle begin_value,
+                   ShapeHandle sizes_value,
+                   std::vector<DimensionHandle>* dims) {
+  for (int i = 0; i < c->Rank(sizes_value); ++i) {
```

Comment:
you can problem inline SliceHelper now and remove it, since it is shorter than before.  Either way is fine though.

Reply:
Thanks @cwhipkey. Done.